lib/perl/Genome/Model/Tools/Vcf/EvaluateVcf.pm

```diff
@@ -0,0 +1,800 @@
+package Genome::Model::Tools::Vcf::EvaluateVcf;
+
+use strict;
+use warnings;
+
+use Genome::File::Vcf::Reader;
+use Genome::File::Vcf::Writer;
+use Genome;
+use Genome::Sys::ShellPipeline;
+use IO::File;
+use IO::Zlib;
+use Path::Class;
+use Genome::Model::Tools::Vcf::VcfCompare;
+
+class Genome::Model::Tools::Vcf::EvaluateVcf {
+    is => "Command::V2",
+    has_input => [
+        reference => {
+            is => 'Genome::Model::Build::ReferenceSequence',
+            doc => 'a reference sequence of interest.',
+        },
+
+        bedtools_version => {
+            is => "Text",
+            doc => "Bedtools version to use",
+            default_value => "2.17.0",
+        },
+
+        joinx_version => {
+            is => "Text",
+            doc => "Joinx version to use",
+            default_value => "1.11",
+        },
+
+        vcflib_version => {
+            is => "Text",
+            doc => "vcflib toolset version to use",
+            default_value => "1.0",
+        },
+
+        vcf => {
+            is => "Text",
+            doc => "VCF file to be evaluated",
+        },
+
+        output_directory => {
+            is => "Text",
+            doc => "Output directory to write to",
+            is_output => 1,
+        },
+
+        old_sample => {
+            is => "Text",
+            doc => "Sample name in the VCF file to use",
+        },
+
+        new_sample => {
+            is => "Text",
+            doc => "Sample name to call the old sample during comparisons",
+        },
+
+        gold_vcf => {
+            is => "Text",
+            doc => "VCF file containing gold standard variant sites "
+                   . "to be used to measure the evaluation VCF"
+        },
+
+        gold_sample => {
+            is => "Text",
+            doc => "Sample name in gold standard vcf file",
+        },
+
+        roi => {
+            is => "Text",
+            doc => "BED file of the target regions to restrict the analysis to",
+        },
+
+        true_negative_bed => {
+            is => "Text",
+            doc => "BED file containing regions where no "
+                   . "variant call should be made",
+        },
+
+        true_negative_size => {
+            is => "Integer",
+            doc => "Use this number as the size of the TN BED "
+                   . "file rather than calculating on the fly",
+            is_optional => '1',
+        },
+
+        pass_only_expression => {
+            is => "Text",
+            doc => "String to pass to vcflib vcffilter in order to "
+                   . "select variants from the evaluation VCF",
+            default_value => q{-g 'FT = PASS | FT = .'},
+        },
+
+        clean_indels => {
+            is => "Text",
+            doc => "Whether or not to exclude indels that don't "
+                   . "actually overlap the ROI",
+            default_value => 0,
+        },
+    ],
+
+    has_output => [
```

Comment:
If this is run from the command line, properties under `has_output` will be listed in the `--help` documentation and will be required to be specified.  If this is undesired `has_transient_optional_output` may be better.  (For background, see also genome/UR@2a0e9e70bbb2b66146393dc03d453ba9a5dd733d.)